flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/StreamExecutionEnvironment.java

```diff
@@ -1786,7 +1796,8 @@ private StateBackend loadStateBackend(ReadableConfig configuration, ClassLoader
 
         SingleOutputStreamOperator<OUT> source =
                 addSource(monitoringFunction, sourceName, null, boundedness)
-                        .transform("Split Reader: " + sourceName, typeInfo, factory);
+                        .transform("Split Reader: " + sourceName, typeInfo, factory)
+                        .setParallelism(readerParallelism);
```

Comment:
I think we'd better not modify the logic when we don't use the adaptive batch scheduler, so I think we also need to add a `readerParallelism == ExecutionConfig.PARALLELISM_DEFAULT && getConfig().isDynamicGraph()` judgment here.

Reply:
agree, I will add a judgment here